### PR TITLE
DOCSP-9558: Fundamentals > CRUD Operations > Read Operations > Limit the Number of Returned Results

### PR DESCRIPTION
**Pull request by ksadoff**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-9558

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodborg-staging.corp.mongodb.com/d2fdb0a/java/docsworker-xlarge/DOCSP-9558/fundamentals/crud/read/limit

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/fundamentals/crud/read/limit.txt
Comment (nathan-contino-mongo) [inline source/fundamentals/crud/read/limit.txt:117]: Should we add a link to this page in a toctree somewhere, or are we holding off on that until we create the CRUD pages that will sit above this page in the sidebar hierarchy?

File: source/fundamentals/crud/read/limit.txt
Comment (nathan-contino-mongo) [inline source/fundamentals/crud/read/limit.txt:49]: Vague method names like this can sometimes cause issues when users try to copy/paste the code, but can't quite get the imports right (particularly since method names like this tend to have overlaps with standard library methods). Consider following the pattern that @ccho-mongodb has used on other docs-java pages to demonstrate imports at the top of an example: https://github.com/mongodb/docs-java/blob/master/source/fundamentals/data-formats/documents.txt#L134

File: source/fundamentals/crud/read/limit.txt
Comment (nathan-contino-mongo) [inline source/fundamentals/crud/read/limit.txt:63]: nitpick: "prints out" might be a more descriptive phrase than "outputs"

File: source/fundamentals/crud/read/limit.txt
Comment (nathan-contino-mongo) [inline source/fundamentals/crud/read/limit.txt:45]: It looks like the Node.js equivalent of this page uses emphasize-lines to highlight the line containing the sort call. I think the code samples is just long enough that this highlight is helpful, but it's a bit of a subjective call. What do you think?

File: source/fundamentals/crud/read/limit.txt
Comment (nathan-contino-mongo) [inline source/fundamentals/crud/read/limit.txt:112]: Couple of things here.
1. Passive voice, suggest reword to active
2. "returned in a Java string format" isn't strictly true here, because we're actually printing the documents and they're turned into a string either way -- toJson() just turns the objects into a different format first, so they use a different toString().
3. Is there a reason why we use toJson() here anyway? I suppose it is a bit more readable than the default toString method, but I wonder if we're better off displaying the results that way instead if we need to include a note anyway to explain what's going on with toJson().

I'd suggest a rewording to something like this, but feel free to reword however you see fit -- this is just a suggestion I came up with on the fly:
```
These results appear in JSON format in this example because
we use the ``toJson()`` method to transform the results into
a JSON format that prints as a JSON string. If you omit the
``toJson()`` method call, the results instead print using the
``toString()`` method of the
:doc:`data format </fundamentals/data-formats>` used to
represent objects in your ``MongoCollection``, which
defaults to ``Document``.
```

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM with some small nitpicks and suggestions.

File: source/fundamentals/crud/read/limit.txt
Comment (ksadoff) [reply source/fundamentals/crud/read/limit.txt:45]: I can add it, I agree that it might be helpful.

File: source/fundamentals/crud/read/limit.txt
Comment (ksadoff) [reply source/fundamentals/crud/read/limit.txt:112]: To your 3rd point, I used it because of readability, but it seems like it might simplify things a bit to simply omit ``toJson()`` and this note altogether, and display the documents as they're originally printed.

**Review by nathan-contino-mongo — COMMENTED** [review]
This looks good, though if we're going to phase out ``toJson()`` calls, I think it's a good idea to also remove the calls from the Sort page.

**Comment by ksadoff** [comment]
> This looks good, though if we're going to phase out `toJson()` calls, I think it's a good idea to also remove the calls from the Sort page.

I had the same thought! I'll make sure to go through that PR and remove those calls (as well as add the relevant ``toctree`` lines in there)

**Comment by nathan-contino-mongo** [comment]
For the sake of merge conflict sanity, it might be a good idea to just include the toctree changes in this PR, then merge the sort PR, then merge master into this PR -- that way, you don't have to fix a merge conflict including the exact same changes in both branches. But honestly just do whatever you feel comfy with.

**Review by ccho-mongodb — COMMENTED** [review]
Let's use 4-space indentation for all Java code examples which is the common formatting convention.

Make sure the antecedent is clear whether in the same sentence or in a larger scope:
E.g. "The limit operation functions as a cap..." vs "The limit operation caps..."
Another e.g. "passing the number of documents to skip over... to find()"
Larger scope example: "Follow the instructions..." but after the insert, the imperative language (do/run this) is omitted.

Try to limit number of details in each sentence (part of our style guide https://docs.mongodb.com/meta/style-guide/writing/write-clear-concise-sentences-paragraphs/).

Similar to the other PR, make sure to consistently use `()` after method names. I see that you've used that syntax (e.g. line 87 and 89), but not in all instances (such as line 42 and 43).

Avoid introducing terms without immediately explaining them. E.g. line 120 "slices".

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Looks pretty good, just added a couple suggestions and comments.

**Review by ccho-mongodb — APPROVED** [review]
LGTM!

**Comment by ksadoff** [comment]
Added @jyemin and @rozza as reviewers, but only one review is necessary

File: source/fundamentals/crud/read/limit.txt
Comment (jyemin) [inline source/fundamentals/crud/read/limit.txt:88]: Given the recent activity on DOCS-9994  and DOCSP-13487 , I'm not sure we should be recommending this sort of usage of skip/limit.  Do we do this in other places?

File: source/fundamentals/crud/read/limit.txt
Comment (jyemin) [inline source/fundamentals/crud/read/limit.txt:113]: @ccho-mongodb this is probably something you should review, since it's the sort of thing we should standardize on across all of our documentation.

Comment (ccho-mongodb) [reply source/fundamentals/crud/read/limit.txt:113]: Sure, talked about this with Katie earlier today. Thanks for bringing it up, I didn't know about this before then. I created a ticket for DevEd to update Node.js documentation and will need to discuss with the driver engineers for the rest of the documentation. I'll have to look into how Server Docs team wants to provide messaging for this as well.

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**
